test/llvm-intrinsics/llvm.umul.with.overflow.ll

```diff
@@ -0,0 +1,91 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis %t.rev.bc -o - | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: Name [[NAME_UMUL_FUNC_8:[0-9]+]] "spirv.llvm_umul_with_overflow_i8"
+; CHECK-SPIRV: Name [[NAME_UMUL_FUNC_32:[0-9]+]] "spirv.llvm_umul_with_overflow_i32"
+; CHECK-SPIRV: Name [[NAME_UMUL_FUNC_VEC_I64:[0-9]+]] "spirv.llvm_umul_with_overflow_v2i64"
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir"
+
+; CHECK-LLVM: [[UMUL_8_TY:%structtype]] = type { i8, i1 }
+; CHECK-LLVM: [[UMUL_32_TY:%structtype.[0-9]+]] = type { i32, i1 }
+; CHECK-LLVM: [[UMUL_VEC64_TY:%structtype.[0-9]+]] = type { <2 x i64>, <2 x i1> }
+
+; On LLVM level, we'll check that the intrinsics were generated again in reverse translation,
+; replacing the SPIR-V level implementations.
+; CHECK-LLVM-NOT: declare {{.*}} @spirv.llvm_umul_with_overflow_{{.*}}
+
+; Function Attrs: nofree nounwind writeonly
+define dso_local spir_func void @_Z4foo8hhPh(i8 zeroext %a, i8 zeroext %b, i8* nocapture %c) local_unnamed_addr #0 {
+entry:
+  ; CHECK-LLVM: call [[UMUL_8_TY]] @llvm.umul.with.overflow.i8
+  ; CHECK-SPIRV: FunctionCall [[#]] [[#]] [[NAME_UMUL_FUNC_8]]
+  %umul = tail call { i8, i1 } @llvm.umul.with.overflow.i8(i8 %a, i8 %b)
+  %cmp = extractvalue { i8, i1 } %umul, 1
+  %umul.value = extractvalue { i8, i1 } %umul, 0
+  %storemerge = select i1 %cmp, i8 0, i8 %umul.value
+  store i8 %storemerge, i8* %c, align 1, !tbaa !2
+  ret void
+}
+
+; CHECK-SPIRV: Function [[#]] [[NAME_UMUL_FUNC_8]]
+; CHECK-SPIRV: FunctionParameter [[#]] [[VAR_A:[0-9]+]]
+; CHECK-SPIRV: FunctionParameter [[#]] [[VAR_B:[0-9]+]]
+; CHECK-SPIRV: IMul [[#]] [[MUL_RES:[0-9]+]] [[VAR_A]] [[VAR_B]]
+; CHECK-SPIRV: UDiv [[#]] [[DIV_RES:[0-9]+]] [[MUL_RES]] [[VAR_A]]
+; CHECK-SPIRV: INotEqual [[#]] [[CMP_RES:[0-9]+]] [[VAR_A]] [[DIV_RES]]
+; CHECK-SPIRV: Select [[#]] [[SELECT_RES:[0-9]+]] [[CMP_RES]]
+; CHECK-SPIRV: CompositeInsert [[#]] [[INSERT_RES:[0-9]+]] [[MUL_RES]]
+; CHECK-SPIRV: CompositeInsert [[#]] [[INSERT_RES_1:[0-9]+]] [[SELECT_RES]] [[INSERT_RES]]
+; CHECK-SPIRV: ReturnValue [[INSERT_RES_1]]
+
+; Function Attrs: nofree nounwind writeonly
+define dso_local spir_func void @_Z5foo32jjPj(i32 %a, i32 %b, i32* nocapture %c) local_unnamed_addr #0 {
+entry:
+  ; CHECK-LLVM: call [[UMUL_32_TY]] @llvm.umul.with.overflow.i32
```

Comment:
Please add check that there is a call of replacement function in SPIR-V module. The same comment applies for `@umulo_v2i64`

Reply:
If I understood you correctly, these checks are already present on [lines 7-9](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/743/files#diff-53b5d620593beaa14591098d7df1b5f8fba7dbdb0a68a382c0d11c0cae94ef1aR7-R9)

Reply:
>If I understood you correctly, these checks are already present on lines 7-9

No, I mean that you have a check that `call` is generated in LLVM IR, but you don't have a check that `FunctionCall` is generated in SPIR-V - just like it is checked in `_Z4foo8hhPh` for `llvm.umul.with.overflow.i8`

Reply:
Oh, my bad, I missed these cases. Uploaded new patch with a fix